### ISSUE TITLE
A call to PInvoke function has unbalanced the stack

### ISSUE THREAD

**Issue by Gerardo-Sista** [DerekGn/NativeUsbLib#11]
Hello Derek,
I hope you're fine.  Sorry for disturbing you again but unfortunately I have this exception: A call to PInvoke function **'NativeUsbLib!NativeUsbLib.WinApis.UsbApi::SetupDiOpenDevRegKey'** has unbalanced the stack.

To see exception you should set like this (standard VS exceptions):
![image](https://user-images.githubusercontent.com/34858800/64123380-8028d100-cda4-11e9-96a0-7ad3c9c6afe9.png)

**This happens when getting the Com port property**. 
Maybe the problem lies in the calling convention, as described here:

https://stackoverflow.com/questions/2941960/a-call-to-pinvoke-function-has-unbalanced-the-stack

Please see the attached example and click on "Exception Code" to see the issue.

[USBdevices.zip](https://github.com/DerekGn/NativeUsbLib/files/3566522/USBdevices.zip)

**Comment by DerekGn** [comment]
I will take a look.

**Comment by DerekGn** [comment]
I suspected it was not the calling convention, it is correct with the default of stdcall aka winapi.

Changing it as in the SO post would actually cause other errors in future. SO answers are not all ways correct.

The issue was the definition of the signature SetupDiOpenDevRegKey
https://docs.microsoft.com/en-us/windows/win32/api/setupapi/nf-setupapi-setupdiopendevregkey

I did not map the REGSAM parameter type correctly, it should have been uint not ulong.

See #12 

The exception is only thrown in debug.  Build will be on nuget shortly.
